helpers/check_format_shell.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module to run shellcheck on all shell scripts.
+
+Written in Python instead of Bash to support an efficient filter of ignored
+files.
+"""
+
+import pathlib
+import subprocess
+
+
+REPO_ROOT = pathlib.Path(__file__).parent.parent.absolute()
+
+ignored_filenames = set((
```

Comment:
I don't know if we should have a separate exclusion list just for shellcheck?

Even if we do, I'd suggest:
1. Keeping this in a separate file, not inline in the script
2. Excluding directories/wildcards, not specific filenames.

Reply:
> Keeping this in a separate file, not inline in the script

What difference does it make where it is?

> Excluding directories/wildcards, not specific filenames.

No.

If someone adds a new Bash script, I want this linter to run, regardless of if they have opted out other files in the past.

Reply:
> What difference does it make where it is?

Config & code should be kept separate, that's a pretty fundamental software development idea. We shouldn't entangle config changes with logical changes.

Since it's a reasonably small number of files, I guess I'm fine with listing specific filenames.